sentinelhub/evalscript.py

```diff
@@ -0,0 +1,118 @@
+    """
+
+    requested_bands = []
+    band_info_dict = {band_info.name: band_info for band_info in data_collection.bands + data_collection.metabands}
+
+    for band_name in bands:
+        if band_name not in band_info_dict:
+            raise ValueError(
+                f"Data collection {data_collection} does not have specifications for band {band_name}.\n"
+                f"Available bands are:\n{[band.name for band in data_collection.bands]}\nand meta-bands:\n"
+                f"{[band.name for band in data_collection.metabands]}."
+            )
+        requested_bands.append(band_info_dict[band_name])
+
+    return requested_bands
+
+
+def generate_evalscript(
+    data_collection: DataCollection,
+    bands: Optional[List[str]] = None,
+    meta_bands: Optional[List[str]] = None,
+    merged_output: Optional[str] = None,
+    prioritize_dn: bool = True,
+) -> str:
+    """Generate an evalscript based on the provided specifications. This utility supports generating only evalscripts
+    with the mosaicking option set to `SIMPLE`.
+
+    :param data_collection: A collection of requested satellite data.
+    :param bands: A list of band names to use in the evalscript. Defaults to using all bands provided by the collection.
+    :param meta_bands: A list of meta band names to use in the evalscript. By default no meta bands are added.
+    :param merged_output: If provided, bands will be concatenated into a single multi-band tiff with this name.
+    :param prioritize_dn: Use DN units if possible. Default is True. If DN units are not available, the default units
+        for each specific band are used. DN units will be used regardless of the flag if they are the only possible
+        choice.
+    """
+
+    band_names = bands if bands is not None else [band.name for band in data_collection.bands]
+    meta_band_names = meta_bands if meta_bands is not None else []
+
+    input_names, input_units = [], []
+    sample_type_map: Dict[str, str] = {}
+    requested_bands = parse_data_collection_bands(data_collection, band_names + meta_band_names)
+
+    for band in requested_bands:
+        unit_choice_idx = band.units.index(Unit.DN) if (prioritize_dn and Unit.DN in band.units) else 0
+        sample_type_map[band.name] = DTYPE_TO_SAMPLE_TYPE[band.output_types[unit_choice_idx]]
+
+        input_names.append(f'"{band.name}"')
+        input_units.append(f'"{band.units[unit_choice_idx].value}"')
+
+    output_spec, return_spec = [], []
+    if merged_output is not None:
+        sample_type = sample_type_map[band_names[0]]
+        output_spec.append(f'{{id: "{merged_output}", bands: {len(band_names)}, sampleType: "{sample_type}"}}')
+        return_spec.append(f"{merged_output}: [{', '.join(f'sample.{band_name}' for band_name in band_names)}]")
+
+    for band_name, sample_type in sample_type_map.items():
+        if merged_output is not None and band_name in band_names:
+            continue
```

Comment:
you can now switch the `for` and `if` statement and get rid of the `continue` right? In fact it can be an `else`.

Reply:
This part of the code has to take care of the bands as well as meta-bands, it just excludes bands if the merged output is provided.

But otherwise you're right, if I change the condition, I could still get rid of the `continue`, if I do it like this:

```python
for band_name, sample_type in sample_type_map.items():
    if merged_bands_output is None or band_name in meta_band_names:
        output_spec.append(f'{{id: "{band_name}", bands: 1, sampleType: "{sample_type}"}}')
        return_spec.append(f"{band_name}: [sample.{band_name}]")
```

Reply:
added a comment, let me know if it's clearer now